From: Ilya Nikolaevskiy <ilnik@webrtc.org>
Date: Tue, 11 Nov 2025 13:25:42 +0100
Subject: (cherry-pick-branch-heads/7499) [M143] Add more logging in
 WgcScreenCaputerer

Original change's description:
> Add more logging in WgcScreenCaputerer
>
> Bug: chromium:448881311
> Change-Id: I08cbe4bf31bcbf99c2503dc27154c82ddd9961bb
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/424700
> Reviewed-by: Henrik Andreassson <henrika@webrtc.org>
> Commit-Queue: Henrik Andreassson <henrika@webrtc.org>
> Auto-Submit: Ilya Nikolaevskiy <ilnik@webrtc.org>
> Commit-Queue: Ilya Nikolaevskiy <ilnik@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46161}

(cherry picked from commit 15b412915e2acc0d4fed7957170f98840cf48543)

Bug: chromium:459774562,chromium:448881311
Change-Id: I08cbe4bf31bcbf99c2503dc27154c82ddd9961bb
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/425140
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: Chrome Cherry Picker <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7499@{#1}
Cr-Branched-From: 1dfd30cf31a029cd127674a11e65c54c07afaf5d-refs/heads/main@{#46046}
---
 .../desktop_capture/win/wgc_capture_source.cc | 41 ++++++++++++++-----
 1 file changed, 31 insertions(+), 10 deletions(-)

diff --git a/modules/desktop_capture/win/wgc_capture_source.cc b/modules/desktop_capture/win/wgc_capture_source.cc
index 86444cb3a6..005be5887b 100644
--- a/modules/desktop_capture/win/wgc_capture_source.cc
+++ b/modules/desktop_capture/win/wgc_capture_source.cc
@@ -21,6 +21,7 @@
 #include "modules/desktop_capture/desktop_geometry.h"
 #include "modules/desktop_capture/win/screen_capture_utils.h"
 #include "modules/desktop_capture/win/window_capture_utils.h"
+#include "rtc_base/logging.h"
 #include "rtc_base/win/get_activation_factory.h"
 
 using Microsoft::WRL::ComPtr;
@@ -134,24 +135,32 @@ bool WgcWindowSource::FocusOnSource() {
 
 HRESULT WgcWindowSource::CreateCaptureItem(
     ComPtr<WGC::IGraphicsCaptureItem>* result) {
-  if (!ResolveCoreWinRTDelayload())
+  if (!ResolveCoreWinRTDelayload()) {
+    RTC_LOG(LS_ERROR) << "Failed to load CoreWinRT library";
     return E_FAIL;
+  }
 
   ComPtr<IGraphicsCaptureItemInterop> interop;
   HRESULT hr = GetActivationFactory<
       IGraphicsCaptureItemInterop,
       RuntimeClass_Windows_Graphics_Capture_GraphicsCaptureItem>(&interop);
-  if (FAILED(hr))
+  if (FAILED(hr)) {
+    RTC_LOG_ERR(LS_ERROR) << "GetActivationFactory failed with hr:" << hr;
     return hr;
+  }
 
   ComPtr<WGC::IGraphicsCaptureItem> item;
   hr = interop->CreateForWindow(reinterpret_cast<HWND>(GetSourceId()),
                                 IID_PPV_ARGS(&item));
-  if (FAILED(hr))
+  if (FAILED(hr)) {
+    RTC_LOG_ERR(LS_ERROR) << "CreateForWindow failed with hr: " << hr;
     return hr;
+  }
 
-  if (!item)
+  if (!item) {
+    RTC_LOG(LS_ERROR) << "CreateForWindow returned null IGraphicsCaptureItem";
     return E_HANDLE;
+  }
 
   *result = std::move(item);
   return hr;
@@ -196,32 +205,44 @@ bool WgcScreenSource::IsCapturable() {
 
 HRESULT WgcScreenSource::CreateCaptureItem(
     ComPtr<WGC::IGraphicsCaptureItem>* result) {
-  if (!hmonitor_)
+  if (!hmonitor_) {
+    RTC_LOG(LS_ERROR) << "hmonitor_ is NULL";
     return E_ABORT;
+  }
 
-  if (!ResolveCoreWinRTDelayload())
+  if (!ResolveCoreWinRTDelayload()) {
+    RTC_LOG(LS_ERROR) << "Failed to load CoreWinRT library";
     return E_FAIL;
+  }
 
   ComPtr<IGraphicsCaptureItemInterop> interop;
   HRESULT hr = GetActivationFactory<
       IGraphicsCaptureItemInterop,
       RuntimeClass_Windows_Graphics_Capture_GraphicsCaptureItem>(&interop);
-  if (FAILED(hr))
+  if (FAILED(hr)) {
+    RTC_LOG_ERR(LS_ERROR) << "GetActivationFactory failed with hr:" << hr;
     return hr;
+  }
 
   // Ensure the monitor is still valid (hasn't disconnected) before trying to
   // create the item. On versions of Windows before Win11, `CreateForMonitor`
   // will crash if no displays are connected.
-  if (!IsMonitorValid(hmonitor_.value()))
+  if (!IsMonitorValid(hmonitor_.value())) {
+    RTC_LOG(LS_ERROR) << "Monitor is invalid";
     return E_ABORT;
+  }
 
   ComPtr<WGC::IGraphicsCaptureItem> item;
   hr = interop->CreateForMonitor(*hmonitor_, IID_PPV_ARGS(&item));
-  if (FAILED(hr))
+  if (FAILED(hr)) {
+    RTC_LOG_ERR(LS_ERROR) << "CreateForMonitor failed with hr: " << hr;
     return hr;
+  }
 
-  if (!item)
+  if (!item) {
+    RTC_LOG(LS_ERROR) << "CreateForMonitor returned null IGraphicsCaptureItem";
     return E_HANDLE;
+  }
 
   *result = std::move(item);
   return hr;
